addons/mail/static/src/core/common/store_service.js

```diff
@@ -364,7 +364,13 @@ export class Store extends BaseStore {
                         insertData.push(vals);
                     }
                 }
-                res[modelName] = store[modelName].insert(insertData, options);
+                const records = store[modelName].insert(insertData, options);
+                if (!res[modelName]) {
+                    res[modelName] = records;
+                } else {
+                    const knownRecordIds = new Set(res[modelName].map((r) => r.localId));
+                    res[modelName].push(...records.filter((r) => !knownRecordIds.has(r.localId)));
+                }
```

Comment:
Why not compare records with `.eq()`? if `.eq()` works, I prefer this over `.localId` comparison

Reply:
This would be n² as using Set is not possible with `eq`.

But if Set could contain raw record then it would be a little easier without local id.

Reply:
Ok, not a big deal all things considered ;)